src/components/app-icon-button/AppIconButton.tsx

```diff
@@ -0,0 +1,15 @@
+import { IconButton, IconButtonProps } from '@mui/material'
+import { FC } from 'react'
+import { Link, LinkProps } from 'react-router-dom'
+
+type AppIconButtonProps = IconButtonProps & Partial<LinkProps>
+
+const AppIconButton: FC<AppIconButtonProps> = ({ to, ...props }) => {
+  if (to) {
+    return <IconButton component={Link} to={to} {...props} />
```

Comment:
Maybe would be better do something like this
```
const AppIconButton: FC<AppIconButtonProps> = ({ to, ...props }) => (
  <IconButton component={to ? Link : 'button'} to={to} {...props} />
)
```

Reply:
Thanks, used your approach so repeating `<IconButton/>` can be avoided